### PR TITLE
feat: improve serverurl interaction in omnibar

### PR DESCRIPTION
**Pull request by antlio**
this pr introduces base url selection from the client app package address bar:

image

### CODE REVIEW

**Review by amritk — APPROVED** [review]
lets gogogo, will iterate on this

**Comment by amritk** [comment]
Alright this PR has been highjacked and its ready for re-review

File: packages/oas-utils/src/entities/workspace/server/server.ts [hunk 1]
```diff
@@ -1,7 +1,43 @@
...
-  url: string
-}
+import { nanoidSchema } from '../shared'
+
+/**
+ * Server variables
+ *
+ * @see https://spec.openapis.org/oas/v3.1.0#server-variable-object
+ */
+export type ServerVariable = z.TypeOf<typeof serverVariableSchema>
```
Comment (amritk) [inline packages/oas-utils/src/entities/workspace/server/server.ts:10]: Copied from OAS type definitions, looks great on hover

**Review by marclave — APPROVED** [review]
awesome work @amritk !

**Pull request merged**